### DISCUSSION TITLE
Quick fix to auto fill annotations from parent class

### DISCUSSION THREAD

**Discussion by corbane** [microsoft/pylance-release#2700] [Enhancement]
## Please search existing feature requests to avoid creating duplicates.

Surprisingly, I did not find any questions on this topic

## Describe the feature you'd like.

When a function is overloaded inside a subclass, the parameters have an Any type

![image](https://user-images.githubusercontent.com/8430545/143183435-6d142114-cf26-412c-b48e-50d3d66e51b1.png)

## Use case

it would be useful to have this functionality. In my case, I am developing a Blender plugin and I am using [fake-bpy-module](https://github.com/nutti/fake-bpy-module) for types.
All the additions of features in the Blender software (Event, Ui, Command, GPU draw, ...) are done by overloading the base classes.
this makes the use of pylance of little use as all types are Any.
It is possible to rewrite the signature, but it's a shame and some functions have complex types, like dict which include tuples including union lists, etc.

---

Thanks for what you do

**Comment by erictraut** [comment]
Parameters with no annotation are always assumed to have an `Any` type. This is standard behavior for Python type checkers. I don't think we'd want to deviate from standard Python behaviors in this regard.

Pylance attempts to help by copying the signature for you when you define a method override in a subclass. For example, if you were to type `def fn` within a new subclass of `A`, Pylance will offer to auto-complete, copying the full signature including type annotations.

![Screen Shot 2021-11-23 at 10 43 37 PM](https://user-images.githubusercontent.com/7040122/143188180-0dea4f38-0ac2-4411-a271-339101232a4d.png)

After accepting the auto-complete:

![Screen Shot 2021-11-23 at 10 43 44 PM](https://user-images.githubusercontent.com/7040122/143188201-48baca8e-0639-4eb5-822d-a981fc665b2b.png)

**Comment by erictraut** [comment]
For method overrides that are already defined, perhaps it would make sense to offer a code action that adds any missing type annotations from the parent class.

**Comment by corbane** [comment]
Hello @erictraut, thank you for the answer

`Pylance will offer to auto-complete`

Yes I use it, but in case of previous scripts or external, it must be rewritten.

I really like Typescript, without the propagation of types it would be difficult to understand and verify old scripts (or parts of code written by someone else.).
JavaScript doesn't even have type annotation defined in its specification, and that doesn't bother the TypeScript team.
I'm surprised Pylance wants to be like other tools.

```perhaps it would make sense to offer a code action```

Yes I think it could be a flexible solution.
The only constraint is that with this we have to rewrite (apply the action code) on the external script parts to have type checking.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
